Diplom3

moiseenko-e opened the pull request.

moiseenko-e committed:
First commit - tests

moiseenko-e committed:
Second commit - report

moiseenko-e committed:
Commit with pom.xml

moiseenko-e committed:
Update README.md

pavel-ulan commented on src/main/java/praktikum/pageobject/LoginPage.java:
```diff
@@ -0,0 +1,66 @@
+    private final By inputPassword = By.xpath("//input[@type='password']"); // проверен
+    private final By enterButton = By.xpath("//*[text()='Войти']"); // проверен
+    private final By registerButton = By.xpath("//*[text()='Зарегистрироваться']"); // проверен
+    private final By forgotPasswordButton = By.xpath("/html/body/div[1]/div/main/div/div/p[2]/a"); // проверен
```
**можно лучше**
очень очень некрасивый локатор, состоящий из абсолютного пути =/
Старайся не использовать локаторы состоящие только из пути (не важно относительный/абсолютный). Как правило всегда можно подобрать другой, более читаемый, локатор. Такие локаторы плохо читаются, по ним непонятно, куда они указывают и если что-то меняется в верстке, то они ломаются и могут или никуда не вести или, что еще хуже, возвращать совершенно другой элемент.

moiseenko-e replied:
исправила

pavel-ulan commented on src/main/java/praktikum/pageobject/LoginPage.java:
```diff
@@ -0,0 +1,66 @@
+    private final By enterButton = By.xpath("//*[text()='Войти']"); // проверен
+    private final By registerButton = By.xpath("//*[text()='Зарегистрироваться']"); // проверен
+    private final By forgotPasswordButton = By.xpath("/html/body/div[1]/div/main/div/div/p[2]/a"); // проверен
+    private final By enterGetText = By.xpath("//h2[contains(text(),'Вход')]"); // проверен
```
**можно лучше**
а локаторы использующие текст могут не работать если у тебя есть несколько локализаций =)
(тут это не актуально, но просто знай)

pavel-ulan commented on src/test/java/praktikum/LoginPageTest.java:
```diff
@@ -0,0 +1,102 @@
+        driver = new ChromeDriver(options);
+        driver.navigate().to(BaseData.BASE_URL);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(WebDriverConfig.WAIT_SECONDS_TIMEOUT));
+        // driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
```
?

moiseenko-e replied:
исправила

pavel-ulan replied:
не исправила =)

pavel-ulan commented on src/test/java/praktikum/LoginPageTest.java:
```diff
@@ -0,0 +1,102 @@
+    @DisplayName("Проверка входа по кнопке 'Войти в аккаунт' на главной странице")
+    public void loginHomePageTest() {
+        objMainPage.clickLogInYourAccountButton();
+        objloginPage.login("test0202@yandex.ru", "012345");
```
**можно лучше**
хардкодишь тестовые данные

pavel-ulan commented on src/test/java/praktikum/LoginPageTest.java:
```diff
@@ -0,0 +1,102 @@
+        objMainPage.clickPersonalAccountButton();
+        objProfilePage.expectationAccountButton();
+        String newAccountLinkText = objProfilePage.getAccountLinkText();
+        assertThat(newAccountLinkText, is("Профиль"));
```
**надо исправить**
а вот тут хардкод вообще недопустим (и нелогичен если честно).
У тебя есть какое-то значение, по которому ты определяешь прошел тест или нет. Логично или вынести это в константу класса как VALID_что-то или (что еще лучше) зашить это в классе шагов или PO как проверку на валидность, т.к. по сути это константное значение для страницы (это не номер заказа или ID пользователя который может меняться, а постоянное поле).

moiseenko-e replied:
исправила, вынесла в константу

pavel-ulan replied:
не вынесла =)

pavel-ulan commented on src/test/java/praktikum/MainPageTests.java:
```diff
@@ -0,0 +1,60 @@
+        objHomePage.clickFillingButton();
+        objHomePage.clickBunButton();
+        String newBreadText = objHomePage.getBunText();
+        assertThat(newBreadText, is("Булки"));
```
**надо исправить**
А тут у тебя вообще тест работает неправильно. (хотя это неочевидно, но если ты проскролишь после перехода к разделу в дебаге и остановишь выполнение, а потом перейдешь в другой раздел, то после продолжения исполнения тест пройдет).
В общем ты проверяешь текст в разделах, но метод проверки на самом деле считает, что текст виден, даже если он проскролен, т.к. этот метод проверяет что у локатора отсутствует свойство hidden.

Попробуй проскролить разделы и смотреть как меняются имена классов в верстке у разделов. И построй проверку "в каком разделе я нахожусь" на этой основе.

moiseenko-e replied:
исправила указанный тест

pavel-ulan replied:
У тебя обе ветки слились в одну. Часть тестов задублировалась, соответственно часть старых и часть новых. Разберись с коммитом пожалуйста.

pavel-ulan reviewed (changes requested):
Тут как-то похуже. Есть неправильные тесты, проверки в остальных с хардкодом, что недопустимо. В общем в этом задании работы будет побольше.

moiseenko-e committed:
Commit changes

Исправлен локатор в элементе кнопка "Восстановить пароль" в классе LoginPage ;
Вынесено значение, по которому я проверяю прохождение теста,  в константу класса LoginPageTest ;
Создан метод для получения значения атрибута элемента таба в классе MainPage, исправлен тест bunTest в классе MainPageTests

pavel-ulan commented on MainPageTests.java:
```diff
@@ -0,0 +1,68 @@
+        assertThat(newBreadText, is("Булки"));
+
+        String attributeValue = objHomePage.getAttributeValueClassFromParentByElement("Булки");
+        assertThat(attributeValue, containsString("tab_tab_type_current"));
```
**надо исправить**
уже лучше. Но у тебя остался прошлый assert, который по сути тут не нужен. А так же у тебя все еще хардкод элемента в тестовом методе, что недопустимо в дипломной работе.
Вынеси проверку, что у текущего элемента есть в локаторе "tab_tab_type_current" в step метод в PO классе. тогда это будет валидно.

pavel-ulan commented on MainPageTests.java:
```diff
@@ -0,0 +1,68 @@
+    public void sauceTest() {
+        objHomePage.clickSauceButton();
+        String newSauceText = objHomePage.getSauceText();
+        assertThat(newSauceText, is("Соусы"));
```
**надо исправить**
в Булках ты поправила, молодец, а тут тест все еще работает так же (точнее не работает).
Всегда проверяй на аналогичные ошибки всю работу

pavel-ulan commented on MainPageTests.java:
```diff
@@ -0,0 +1,68 @@
+    public void fillingTest() {
+        objHomePage.clickFillingButton();
+        String newFillingText = objHomePage.getFillingText();
+        assertThat(newFillingText, is("Начинки"));
```
**надо исправить**
что бы не потеряла - тут тоже

pavel-ulan commented on src/main/java/praktikum/pageobject/MainPage.java:
```diff
@@ -0,0 +1,88 @@
+
+public class MainPage {
+    private final WebDriver driver;
+    private final By personalAccountButton = By.xpath("/html/body/div/div/header/nav/a/p"); // проверен
```
**надо исправить**
а этот локатор под определение "абсолютный путь" не подпадает разве?

pavel-ulan commented on src/test/java/praktikum/LoginPageTest.java:
```diff
@@ -0,0 +1,102 @@
+        objMainPage.clickPersonalAccountButton();
+        objProfilePage.expectationAccountButton();
+        String newAccountLinkText = objProfilePage.getAccountLinkText();
+        assertThat(newAccountLinkText, is("Профиль"));
```
и тут

pavel-ulan commented on src/test/java/praktikum/LoginPageTest.java:
```diff
@@ -0,0 +1,102 @@
+        objMainPage.clickPersonalAccountButton();
+        objProfilePage.expectationAccountButton();
+        String newAccountLinkText = objProfilePage.getAccountLinkText();
+        assertThat(newAccountLinkText, is("Профиль"));
```
и тут

pavel-ulan commented on src/test/java/praktikum/LoginPageTest.java:
```diff
@@ -0,0 +1,102 @@
+        objMainPage.clickPersonalAccountButton();
+        objProfilePage.expectationAccountButton();
+        String newAccountLinkText = objProfilePage.getAccountLinkText();
+        assertThat(newAccountLinkText, is("Профиль"));
```
и тут

pavel-ulan reviewed (changes requested):
У тебя обе ветки слились в одну. Часть тестов задублировалась, соответственно часть старых и часть новых. Разберись с коммитом пожалуйста.

moiseenko-e committed:
Delete MainPage.java

moiseenko-e committed:
Изменения

Обновила файлы в классах:
- Вынесла в константу проверку текста "Профиль" в классе LoginPageTest
- Переход по разделам: Вынесла проверку, что у текущего элемента есть в локаторе "tab_tab_type_current" в step метод в PO классе MainPage и исправила тесты в MainPageTests
- Исправила локатор в элементе кнопка "Личный кабинет" в классе MainPage

moiseenko-e committed:
Обновлен отчет

moiseenko-e committed:
Delete LoginPage.java

moiseenko-e committed:
Delete LoginPageTest.java

moiseenko-e committed:
Delete MainPageTests.java

pavel-ulan commented on src/main/java/praktikum/pageobject/LoginPage.java:
```diff
@@ -10,12 +10,12 @@
+    private final By inputPassword = By.xpath("//input[@type='password']");
+    private final By enterButton = By.xpath("//*[text()='Войти']");
+    private final By registerButton = By.xpath("//*[text()='Зарегистрироваться']");
+    private final By forgotPasswordButton = By.xpath("//*[text()='Восстановить пароль']"); // исправлен
```
👍

pavel-ulan commented on src/main/java/praktikum/pageobject/MainPage.java:
```diff
@@ -67,22 +58,49 @@ public String getDesignerText(){
-    public String getSauceText(){
-        return driver.findElement(sauceHeaderText).getText();
+
+   // добавила методы для проверки, что таб не выбран
```
**хорошо**
Да, молодец 👍

pavel-ulan reviewed (approved):
Да, сейчас все действительно будет работать. 👍
